src/ray/util/url.cc

```diff
@@ -16,64 +16,103 @@
 
 #include <stdlib.h>
 
+#ifdef _WIN32
+#include <afunix.h>
+#else
+#include <sys/un.h>
+#endif
+
 #include <algorithm>
-#include <boost/asio/ip/address.hpp>
+#include <boost/asio/generic/stream_protocol.hpp>
+#ifndef _WIN32
+#include <boost/asio/local/stream_protocol.hpp>
+#endif
 #include <boost/asio/ip/tcp.hpp>
+#include <sstream>
 #include <string>
 
+#include "ray/util/filesystem.h"
 #include "ray/util/logging.h"
+#include "ray/util/util.h"
 
-boost::asio::ip::tcp::endpoint parse_ip_tcp_endpoint(const std::string &endpoint,
-                                                     int default_port) {
-  const std::string scheme_sep = "://";
-  size_t scheme_begin = 0, scheme_end = endpoint.find(scheme_sep, scheme_begin);
-  size_t host_begin;
-  if (scheme_end < endpoint.size()) {
-    host_begin = scheme_end + scheme_sep.size();
-  } else {
-    scheme_end = scheme_begin;
-    host_begin = scheme_end;
-  }
-  std::string scheme = endpoint.substr(scheme_begin, scheme_end - scheme_begin);
-  RAY_CHECK(scheme_end == host_begin || scheme == "tcp");
-  size_t port_end = endpoint.find('/', host_begin);
-  if (port_end >= endpoint.size()) {
-    port_end = endpoint.size();
+namespace ray {
+
+std::string endpoint_to_url(
+    const boost::asio::generic::basic_endpoint<boost::asio::generic::stream_protocol> &ep,
+    bool include_scheme) {
+  std::string result;
+  switch (ep.protocol().family()) {
```

Comment:
For this whole block we can just call `ep.address().to_string()`, like this:
```
if (ep.protocol().family() == AF_INET || ep.protocol().family() == AF_INET6) {
    return "tcp://" + ep.address().to_string();
}
```

I'd say let's get rid of the `include_schema` option and always include it in the `endpoint_to_url`, there is not really a reason to not include it, is there?

Reply:
Well the reason to not always include it is that `ClientConnection::RemoteEndpointInfo()` didn't include it before, and I wasn't trying to change its behavior. Should I?

Reply:
Regarding `address()`, it won't work; see my reply to the other comment below.